apps/webapp/app/utils/createHttpSourceRequest.ts

```diff
@@ -0,0 +1,14 @@
+import { HttpSourceRequest } from "@trigger.dev/core";
+import { requestUrl } from "./requestUrl.server";
+
+export async function createHttpSourceRequest(request: Request): Promise<HttpSourceRequest> {
+  const url = requestUrl(request);
+  return {
+    headers: Object.fromEntries(request.headers) as Record<string, string>,
+    url: url.href,
+    method: request.method,
+    rawBody: ["POST", "PUT", "PATCH"].includes(request.method)
```

Comment:
This was taken from the HttpSource code. Is this the best way to deal with a body? 

In theory a `GET` can have a body… also it's possible that a `POST` doesn't have a body, in which case this would throw an error I think. What about `Delete`?

What if instead we just try catch request.arrayBuffer() and if there's data we put it in the `rawBody`? Feels more comprehensive?

@ericallam @nicktrn

Reply:
> What if instead we just try catch request.arrayBuffer() and if there's data we put it in the rawBody?

I think this may have to be done to cover all edge cases. We shouldn't count on external services being sensible.